ext/standard/string.c

```diff
@@ -4926,13 +4926,22 @@ PHP_FUNCTION(setlocale)
 {
 	zend_long cat;
 	zval *args = NULL;
-	int num_args;
+	uint32_t num_args;
 
 	ZEND_PARSE_PARAMETERS_START(2, -1)
 		Z_PARAM_LONG(cat)
 		Z_PARAM_VARIADIC('+', args, num_args)
 	ZEND_PARSE_PARAMETERS_END();
 
+	if (ZEND_ARG_USES_STRICT_TYPES()) {
+		for (uint32_t i = 0; i < num_args; i++) {
+			if (UNEXPECTED(Z_TYPE(args[i]) != IS_ARRAY && Z_TYPE(args[i]) != IS_STRING)) {
+				zend_wrong_parameter_type_error(i + 2, Z_EXPECTED_ARRAY_OR_STRING, &args[i]);
+				RETURN_THROWS();
+			}
+		}
+	}
```

Comment:
This doesn't cover the case where a non-stringable object is passed as argument, nor a resource, or null.

Reply:
That and also types that should be rejected by our normal semantics were not :)